### DISCUSSION TITLE
How to get rid of printed messages in Python using OMPython

### DISCUSSION THREAD

**Discussion by ejperezHI** [OpenModelica/OpenModelica#12041] [Q&A]
Hello everyone,
I am using the "OMPython" library for simulating OpenModelica models in python, and when executing the command:
`mod = ModelicaSystem(modelpath, name_model)`
Or any command involving this variable (such as the .simulate() method):
`mod.simulate()`

Some messages are printed, such as those:
![image](https://github.com/OpenModelica/OpenModelica/assets/131266350/abf71fdf-d006-462a-995f-2dfa3430410b)
![image](https://github.com/OpenModelica/OpenModelica/assets/131266350/b1dce90c-4f11-458f-af24-059f8523488b)

It works all perfect it is just that those prints are uncomfortable. Is there any way of get rid of this uncomfortable prints? 

Thank you a lot in advance,

**Comment by adeas31** [comment]
@arun3688 any hint?

**Comment by arun3688** [comment]
@ejperezHI this is fixed with this commit https://github.com/OpenModelica/OMPython/commit/adf27082fd1a742dc8182a056a55b90c32d0ef4a, please pull the latest changes and try it out, For simulate API you have to use the following 

`simulate(verbose=False)` 

By default all the print messages are shown in the terminal, you have to set it False to avoid those print statements

**Reply by ejperezHI** [reply]
Thanks for your quick reply,

According to the .simulate() method, it doesn't seem to have the option of "verbose=False" within its arguments:
https://build.openmodelica.org/Documentation/OpenModelica.Scripting.simulate.html
https://openmodelica.org/doc/OpenModelicaUsersGuide/manual-update/ompython.html
Also checking it with the inspect" library, it says me that those are the arguments:
['self', 'resultfile', 'simflags']

I guess that the "verbose" argument is updated in that new commit, right? 

Thanks,

**Comment by arun3688** [comment]
@ejperezHI yes you have to pull the changes from the git and then do `python setup.py install `

**Reply by adeas31** [reply]
Yes, if @ejperezHI is using the sources. If he is taking it from the pip then it will not work. The last release was done almost 2 years ago. Maybe its time to make a new OMPython release.

**Reply by arun3688** [reply]
@adeas31 I made a new relase v3.5.0 `https://github.com/OpenModelica/OMPython/releases/tag/v3.5.0` @ejperezHI now you can directly install from pip for the latest changes

**Reply by adeas31** [reply]
@arun3688 you need to publish it to the pip as well https://pypi.org/project/OMPython/

**Reply by arun3688** [reply]
@adeas31 I think I don't have the account to publish in pip atleast for OMPython, can you publish it

**Reply by ejperezHI** [reply]
Thanks all of you for your replies,

Yes, the best would be to publish in the pip, so I can be updated, downloaded and imported the same way always, and to not depend on individual commits

Thanks,

**Reply by arun3688** [reply]
@ejperezHI It is already published to pip so you can install from pip, but we cannot make all minor changes to pip, in most cases you have to use `python -m pip install -U https://github.com/OpenModelica/OMPython/archive/master.zip
`
